### PR TITLE
python312Packages.changefinder: drop nose dependency

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

part of:
- #326513

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/changefinder/default.nix [hunk 1]
```diff
@@ -20,17 +20,22 @@ buildPythonPackage {
     hash = "sha256-1If0gIsMU8673fKSSHVMvDgR1UnYgM/4HiyvZJ9T6VM=";
   };
 
+  patches = [ ./fix_test_invocation.patch ];
+  pythonRemoveDeps = [ "nose" ];
```
Comment (emilazy) [inline pkgs/development/python-modules/changefinder/default.nix:24]: Nit:
```suggestion
  patches = [ ./fix_test_invocation.patch ];
  
  pythonRemoveDeps = [ "nose" ];
```

File: pkgs/development/python-modules/changefinder/default.nix [hunk 2]
```diff
@@ -20,17 +20,22 @@ buildPythonPackage {
...
+
   nativeBuildInputs = [
     setuptools
   ];
 
   propagatedBuildInputs = [
-    nose # not actually required during runtime, but specified as required in `setup.py`
+    pytestCheckHook
```
Comment (emilazy) [inline pkgs/development/python-modules/changefinder/default.nix:31]: This should be in `nativeCheckInputs` instead.

**Review by emilazy — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/changefinder/fix_test_invocation.patch [hunk 3]
```diff
@@ -0,0 +1,18 @@
+diff --git a/test/test.py b/test/test.py
+index 85a9f4e..9150a90 100644
+--- a/test/test.py
++++ b/test/test.py
+@@ -15,11 +15,13 @@ class TestChangeFinder():
+                                      np.random.rand(300)])
+ 
+     def test_changefinder(self):
++        self.setup()
```
Comment (emilazy) [inline pkgs/development/python-modules/changefinder/fix_test_invocation.patch:9]: The [pytest documentation](https://docs.pytest.org/en/7.1.x/how-to/nose.html) suggests that it should automatically be calling the `setup()` method for nose compatibility. Do you have any idea why that’s not working? I’ll try to look into it if not.

Comment (Sigmanificient) [reply pkgs/development/python-modules/changefinder/fix_test_invocation.patch:9]: Didn't knew, not sure

Comment (emilazy) [reply pkgs/development/python-modules/changefinder/fix_test_invocation.patch:9]: Ah, they removed the support: <https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose>.

Could you rename `def setup(self):` (and `def teardown(self):` if any) to `def setup_method(self):`? That should make it just work without having to patch the individual tests.

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 330311` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.changefinder</li>
    <li>python311Packages.changefinder.dist</li>
    <li>python312Packages.changefinder</li>
    <li>python312Packages.changefinder.dist</li>
  </ul>
</details>

**Pull request merged**